projects/plugins/jetpack/_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php

```diff
@@ -1563,21 +1564,17 @@ public function get_stats_data( WP_REST_Request $request ) {
 		) {
 			$range = 'day';
 		}
-
-		if ( ! function_exists( 'stats_get_from_restapi' ) ) {
-			require_once JETPACK__PLUGIN_DIR . 'modules/stats.php';
-		}
-
+		$wpcom_stats = new WPCOM_Stats();
 		switch ( $range ) {
 
 			// This is always called first on page load.
 			case 'day':
-				$initial_stats = stats_get_from_restapi();
+				$initial_stats = json_decode( wp_json_encode( $wpcom_stats->get_stats() ) );
```

Comment:
Let's make sure we check and handle `WP_Error` responses before attempting to encode/decode.
Also let's handle cases where encoding and decoding is not successful.
Applies to all cases here.

Reply:
Thanks for the review. I have made changes to improve error handling.